tests/end_to_end_tests/run_in_remote_machine/cleanup_test_ec2_objects.py

```diff
@@ -0,0 +1,232 @@
+    )
+    entries = resp["Entries"]
+
+    entries_to_remove = {}
+
+    current_time = time.time()
+
+    # Remove old prfix list entries.
+    for entry in entries:
+        timestamp = _parse_entry_timestamp(entry)
+        if timestamp <= current_time - PREFIX_LIST_ENTRY_REMOVE_THRESHOLD:
+            entries_to_remove[entry["Cidr"]] = entry
+
+    # If workflow provided, then we also remove entries that have matching workflow_id field in
+    # their Description field.
+    if workflow_id:
+        for entry in entries:
+            description = _parse_entry_description(entry)
+            if description["workflow_id"] and description["workflow_id"] == workflow_id:
+                entries_to_remove[entry["Cidr"]] = entry
+
+    if not entries_to_remove:
+        return
+
+    print(f"Removing entries: {entries_to_remove}")
+    _remove_entries(
+        client=client,
+        entries=list(entries_to_remove.values()),
+        prefix_list_id=prefix_list_id
+    )
+
+
+def cleanup_old_ect_test_instance(
+        libcloud_ec2_driver,
+        workflow_id: str = None
+):
+    """
+    Cleanup old ec2 test instances.
+    """
+    from libcloud.utils.iso8601 import parse_date
+    nodes = libcloud_ec2_driver.list_nodes()
+
+    print("Looking for and deleting old running automated test nodes...")
+
+    nodes_to_delete = []
+
+    for node in nodes:
+        if INSTANCE_NAME_STRING not in node.name:
+            continue
+
+        # Re remove instances which are created by the current workflow immediately.
+        if workflow_id in node.name:
+            nodes_to_delete.append(node)
+            continue
+
+        launch_time = node.extra.get("launch_time", None)
+
+        if not launch_time:
+            continue
+
+        launch_time_dt = parse_date(launch_time).replace(tzinfo=None)
+        if launch_time_dt >= DELETE_OLD_NODES_THRESHOLD_DT:
+            continue
+
+        print(('Found node "%s" for deletion.' % (node.name)))
+
+        nodes_to_delete.append(node)
+
+    # TODO: For now we only print the node names to ensure script doesn't incorrectly delete
+    # wrong nodes. We should uncomment out deletion once we are sure the script is correct.
+    for node in nodes_to_delete:
+        assert INSTANCE_NAME_STRING in node.name
+        print("")
+        destroy_node_and_cleanup(driver=driver, node=node)
+
+    print("")
+    print("Destroyed %s old nodes" % (len(nodes_to_delete)))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--access-key",
+        required=True
+    )
+    parser.add_argument(
+        "--secret-key",
+        required=True
+    )
+    parser.add_argument(
+        "--prefix-list-id",
+        required=True
+    )
+    parser.add_argument(
+        "--region",
+        required=True
+    )
+    parser.add_argument(
+        "--workflow-id",
+        required=False
+    )
+    args = parser.parse_args()
+
+    boto_client = boto3.client(
```

Comment:
This can be a TODO / future work, is there a way to generalize using libcloud to avoid having to specifically make a boto3 client here?  (I haven't played with it but I figured libcloud would provide generalized classes to make it easy to switch between cloud providers)

Reply:
The `libcloud` does provide generalized interface, but, IFAIK, it does not provide it for AWS prefix lists manipulation, at least I could not find it :(